osbs/utils/__init__.py

```diff
@@ -383,12 +383,29 @@ class ImageName(object):
                       |------------------------| image
     |------------------------------------------| image
     """
+    __DEFAULT = object()  # constant class variable to allow for None assignment
 
     def __init__(self, registry=None, namespace=None, repo=None, tag=None):
-        self.registry = registry
-        self.namespace = namespace
-        self.repo = repo
-        self.tag = tag or 'latest'
+        self.__registry = registry
```

Comment:
I'm not sure if want to use double underscore, it makes inheritance hard. Would be enough to have just one?

Reply:
Would make sense to use frozen dataclass here, that should provide all what we need?